loader/base_loader.py

```diff
@@ -108,6 +118,106 @@ def __init__(self, dss_endpoint: str, staging_bucket: str, google_project_id: st
         dss_config['DSSClient'].swagger_url = f'{self.dss_endpoint}/swagger.json'
         self.dss_client = DSSClient(config=dss_config)
 
+    @staticmethod
+    def get_s3_metadata_client(aws_meta_cred, session='NIH-Test', duration=43199):
+        """
+        Access AWS credentials from a file and supply a client for them.
+
+        :param aws_meta_cred: File containing an AWS ARN for an AssumedRole, e.g.:
+                              'arn:aws:iam::************:role/ROLE_NAME_HERE'
+        :param duration: How long, in seconds, the AssumedRole will be valid for.
+        :return: An AWS s3 client object authorized with the above credentials or None.
+        """
+        if not aws_meta_cred:
+            return None
+
+        client = boto3.client('sts')
+        with open(aws_meta_cred, 'r') as f:
+            rolearn = f.read().strip()
+
+        # DurationSeconds can have a value from 900s to 43200s (as of 10.23.2018).
+        # 900s = 15 min; 43200s = 12 hours
+        # https://docs.aws.amazon.com/cli/latest/reference/sts/assume-role.html
+        print(rolearn)
+        assumed_role = client.assume_role(RoleArn=rolearn, RoleSessionName=session, DurationSeconds=duration)
+
+        credentials = assumed_role['Credentials']
+        return boto3.client('s3',
+                            aws_access_key_id=credentials['AccessKeyId'],
+                            aws_secret_access_key=credentials['SecretAccessKey'],
+                            aws_session_token=credentials['SessionToken'])
+
+    def get_gs_metadata_client(self, gcp_meta_cred):
+        """
+        Access Google credentials from a file and supply a client for them.
+
+        :param gcp_meta_cred: File containing user credentials, usually generated via:
+                                gcloud auth application-default login
+                              Default location:
+                                '/home/<user>/.config/gcloud/application_default_credentials.json'
+        :return: A google storage client object authorized with the above credentials or None.
+        """
+        if not gcp_meta_cred:
+            return None
+
+        credentials = Credentials(token=None).from_authorized_user_file(gcp_meta_cred)
+        return Client(project=self.google_project_id, credentials=credentials)
```

Comment:
If a GCP project is provided, it should also be passed to the default Client (not shown here).
The project is for access to requester pays buckets, and it is legitimate to have a public access bucket that is requester pays.

Reply:
Done.